kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala

```diff
@@ -48,14 +49,18 @@ class KyuubiSessionManager private (name: String) extends SessionManager(name) {
 
     val username = Option(user).filter(_.nonEmpty).getOrElse("anonymous")
 
+    val sessionConf = this.getConf.getUserDefaults(user)
+    if (!sessionConf.get(HA_ZK_ACL_ENGINE_ENABLED)) {
```

Comment:
It‘s a bit hacky, can we use HA_ZK_ACL_ENGINE_ENABLED at the caller side?

Reply:
Move to `ZooKeeperACLProvider`